style_guides/js_style_guide.md

```diff
@@ -608,158 +729,97 @@ function loadUser(id) {
   ...
 }
 
-var isSessionValid = (session.expires < Date.now());
+const isSessionValid = (session.expires < Date.now());
 if (isSessionValid) {
   ...
 }
-```
 
-*Wrong:*
+// bad
 
-```js
 // Execute a regex
-var matches = item.match(/ID_([^\n]+)=([^\n]+)/));
+const matches = item.match(/ID_([^\n]+)=([^\n]+)/));
 
 // Usage: loadUser(5, function() { ... })
 function loadUser(id, cb) {
   // ...
 }
 
 // Check if the session is valid
-var isSessionValid = (session.expires < Date.now());
+const isSessionValid = (session.expires < Date.now());
 // If the session is valid
 if (isSessionValid) {
-  // ...
+  ...
 }
 ```
 
 ## Do not comment out code
 
-We use a version management system. If a line of code is no longer needed, remove it, don't simply comment it out.
+We use a version management system. If a line of code is no longer needed,
+remove it, don't simply comment it out.
 
 ## Classes/Constructors and Inheritance
 
-While JavaScript it is not always considered an object-oriented language, it does have the building blocks for writing object oriented code. Of course, as with all things JavaScript, there are many ways this can be accomplished. Generally, we try to err on the side of readability.
-
-### Capitalized function definition as Constructors
-
-When Defining a Class/Constructor, use the function definition syntax.
-
-*Right:*
-```js
-function ClassName() {
-
-}
-```
-
-*Wrong:*
-```js
-var ClassName = function () {};
-```
-
-### Inheritance should be done with a utility
-
-While you can do it with pure JS, a utility will remove a lot of boilerplate, and be more readable and functional.
-
-*Right:*
+In general, use of the `new` operator is discouraged. If you must use a
```

Comment:
I think we need to explain why `new` is discouraged. What problems does it create? Does it impact testability? Readability? Scalability?

Given that the "good" example below uses `new`, I think we could also just remove this sentence.